frontends/p4/defaultValues.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2023 VMWare, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "defaultValues.h"
+
+namespace P4 {
+
+const IR::Expression *DoDefaultValues::defaultValue(const IR::Expression *expression,
```

Comment:
Possible to make this a general utility function? There is also some overlap with https://github.com/p4lang/p4c/blob/main/ir/irutils.h. We can merge some of this.

Reply:
This is tied to the https://p4.org/p4-spec/docs/P4-16-v-1.2.3.html#sec-default-values default values section of the spec. Where else is this function useful?

Reply:
I have factored out the defaultValue function as a static method of this class.
Hopefully it is what you need.